union/core/core.py

```diff
@@ -0,0 +1,124 @@
+import inspect
+import io
+import sys
+import traceback
+
+import aiohttp
+import discord
+from discord.ext import commands
+
+
+class Core:
+    """
+    Core features of the bot.
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.session = aiohttp.ClientSession(loop=bot.loop)
+
+    def __unload(self):
+        self.session.close()
+
+    @commands.group(invoke_without_command=True)
+    @commands.is_owner()
+    async def change(self, ctx):
+        """
+        Change the looks of the bot.
+        """
+        await ctx.send(f"\N{RIGHT-POINTING MAGNIFYING GLASS} Please provide a valid subcommand.")
+
+    @change.command(aliases=["username"])
+    @commands.is_owner()
+    async def name(self, ctx, *, username: commands.clean_content):
+        """
+        Change the username of the bot.
+        """
+        await ctx.bot.user.edit(username=username)
+        await ctx.send(f"\N{MEMO} Changed username to {username}")
+
+    @change.command(aliases=['playing'])
+    @commands.is_owner()
+    async def game(self, ctx, *, message: commands.clean_content):
+        """
+        Change the game of the bot.
+        """
+        await ctx.bot.change_presence(game=discord.Game(name=message))
+        await ctx.send(f"\N{VIDEO GAME} Changed game to {message}")
+
+    @change.command()
+    @commands.is_owner()
+    async def status(self, ctx, *, status):
+        """
+        Change the status of the bot.
+        """
+        await ctx.bot.change_presence(status=status)
+        await ctx.send("\N{HEAVY LARGE CIRCLE} Changed status.")
+
+    @change.command(aliases=["avy"])
+    @commands.is_owner()
+    async def avatar(self, ctx, url: str = None):
+        """
+        Change the avatar of the bot.
+
+        You can use either attachments or image URLs.
+        Providing no attachment or URL will result in a default avatar.
+        """
+        buffer = io.BytesIO()
+
+        if url is None:
+            attachments = ctx.message.attachments
+            if attachments:
+                await attachments[0].save(buffer)
```

Comment:
Do we want to add size checks here in case someone tries to upload several very large files in quick succession?